pkg/queue/instance_test.go

```diff
@@ -86,3 +86,30 @@ func TestRetry(t *testing.T) {
 	// wait for the task to run twice.
 	wg.Wait()
 }
+
+func TestResourceFree(t *testing.T) {
+	q := NewQueue(1 * time.Microsecond)
+	stop := make(chan struct{})
+	signal := make(chan struct{})
+	go func() {
+		q.Run(stop)
+		signal <- struct{}{}
+	}()
+
+	q.Push(func() error {
+		t.Log("mock exec")
+		return nil
+	})
+
+	// mock queue wait cond signal
+	time.Sleep(200 * time.Millisecond)
```

Comment:
nit: use channel to notify instead of sleep

Reply:
use this instead of sleep, but this pr is closed
```go
// mock queue wait cond signal
time.AfterFunc(time.Millisecond*10, func() {
	// wait task exec complete, block on the `q.cond.Wait()`, and close stop signal
	close(stop)
})
```